src/02-parser/desugarer.ml

```diff
@@ -28,8 +28,10 @@ type state = {
   field_symbols : Type.Field.t StringMap.t;
   tyname_symbols : (TyName.t * int) StringMap.t;
   constructors : (Type.Label.t * constructor_kind) StringMap.t;
-  local_type_annotations : Type.ty StringMap.t;
+  local_type_annotations : (Type.TyParam.t * Type.ty) StringMap.t;
   inlined_types : Type.ty StringMap.t;
+  param_variance : VarianceSet.t Type.TyParam.Map.t;
+  type_application_names : (Type.TyParam.t * variance) list Term.Variable.Map.t;
```

Comment:
A better name would be `type_constructors`, and I guess this should replace `tyname_symbols`, so have a `TyName.t` attached. There is no need for the `int` because you have the length of the parameter list.